module/VuFind/src/VuFind/View/Helper/Root/Component.php

```diff
@@ -67,6 +77,9 @@ public function __invoke(string $name, $params = []): string
 
         ++$invocation;
         $params['_invocation'] = $invocation;
+        // Theme specific class name to prevent accidental style calls. Prefix vc- stands for VuFind component.
+        $normalizedName = str_replace('/', '-', $name);
+        $params['_themeClass'] = 'vc-' . $this->themeInfo->getTheme() . '-' . $normalizedName;
```

Comment:
One more thought: might there be an advantage to putting a `vufind-theme-x` class on the `<body>` tag in the layout, and then leaving the theme name out of the component class? That way, you could target styles for the component in general, or have more specific theme-focused styles. It would make individual class names shorter and less redundant. Or is there a reason it's better to do it this way?

Also, for $normalizedName, might it be better to do a preg_replace so you can replace any characters that are not legal in a class name? e.g. something like `[^-_A-Za-z0-9]`?

Reply:
hmm, that could actually be quite nice. Will adjust!